app/uploaders/attachment_uploader.rb

```diff
@@ -46,8 +46,13 @@ def pdf?(file)
   end
 
   def get_first_page_as_png(width, height)
+    # Open3 has an `ensure` block that catches the `Timeout::Error` raised by `Timeout`, which basically nullifies
+    # the Timeout. An ugly (but one-line) workaround is to put a second `Timeout` around the first.
+    # See: https://bugs.ruby-lang.org/issues/4681
     output, status = Timeout.timeout(THUMBNAIL_GENERATION_TIMEOUT) do
-      Open3.capture2e(pdf_thumbnail_command(width, height))
+      Timeout.timeout(THUMBNAIL_GENERATION_TIMEOUT) do
+        Open3.capture2e(pdf_thumbnail_command(width, height))
+      end
```

Comment:
Wow, nice find! 🎉 

I wonder if this would be simpler if we avoided using `Open3` all together and did something like:

```ruby
output = `#{pdf_thumbnail_command(width, height)`
status = $?.exitstatus
```

That way we wouldn't need nested timeouts, I think?

Reply:
Wow. Didn't even know this was a thing - why were we using `Open3` in the first place then? I'll try it out on integration. Would be great to bin double timeouts.

Reply:
@tuzz done - feels much nicer!